duckduckgo/Android
# Backup service

Task/Issue URL: https://app.asana.com/0/0/1206040339100181/f

### Description
Implemented backup key/value service

### Steps to test this PR

_ATB mapped to BackUp prefs_
- Make sure you have a previous version of the app installed
- Put a log on _Line 35_ of `ReinstallAtbListener` to check preference mapping
- Install from branch
- [x] Check log appears in Logcat

_Backup service_
- Install from branch
- Run `./test_cloud_backup.sh com.duckduckgo.mobile.android.debug` in the terminal
- [x] Check backup is success

_Pixel sent when Backup service enabled_
- Install from this branch
- Run `./test_cloud_backup.sh com.duckduckgo.mobile.android.debug` in the terminal
- Wait until backup is success
- Go to Logcat
- [x] Check `m_backup_service_enabled` pixel is sent


### No UI changes
